mxcube3/ui/components/Tasks/DataCollection.js

```diff
@@ -1,137 +1,290 @@
+            optSAD: fields.optSAD.value,
+            spaceGroup: fields.spaceGroup.value,
+            crystMin: fields.crystMin.value,
+            crystMax: fields.crystMax.value,
+            omegaMin: fields.omegaMin.value,
+            omegaMax: fields.omegaMax.value,
+            stratComp: fields.stratComp.value,
+            point : this.props.pointId
+
+            };
+        if (this.props.sampleIds.constructor == Array){
+
+                this.props.sampleIds.map((sampleId) =>{
+
+                    let queueId = this.props.lookup[sampleId];
+                    if (queueId) {
+                        this.props.addTask(queueId, sampleId, parameters);
                     } else {
-                        // the sample is not in queue yet
-                        this.props.addSampleAndTask(sample_id, parameters);
-                    }                    
-                }
+                    // the sample is not in queue yet
+                    this.props.addSampleAndTask(sampleId, parameters);
+                    }  
             });
+          
+        }else{
+            let sample_queue_id = this.props.lookup[ this.props.sampleIds];
+            this.props.changeTask(this.props.taskData.queue_id, sample_queue_id,this.props.sampleIds, parameters, runNow);
         }
-        
         this.props.hide();
     }
 
 
     render() {
-        var {num_images, exp_time, resolution, osc_start , energy, osc_range, transmission} = this.props.fields; 
- 
-        return (<Modal animation={false} show={this.props.show} onHide={this.props.hide}>
-          <Modal.Header closeButton>
-            <Modal.Title>Data Collection</Modal.Title>
-          </Modal.Header>
-          <Modal.Body>
-        <form className="form-horizontal">
 
-            <div className="form-group">
+        const {fields: {numImages, expTime, resolution, oscStart , energy, oscRange, transmission, centringMethod, detectorMode, kappa, phi, radiationDamage, optSAD, spaceGroup, crystMin, crystMax, omegaMin, omegaMax, stratComp }} = this.props;
+
+        const style = {
+          overlay : {
+            position          : 'fixed',
+            top               : 0,
+            left              : 0,
+            right             : 0,
+            bottom            : 0,
+            backgroundColor   : 'rgba(255, 255, 255, 0.75)'
+        },
+        content : {
+            position                   : 'absolute',
+            top                        : '40px',
+            left                       : '40px',
+            right                      : '40px',
+            bottom                     : '40px',
+            border                     : 'none',
+            background                 : 'none',
+            overflow                   : 'auto',
+            WebkitOverflowScrolling    : 'touch',
+            borderRadius               : '4px',
+            outline                    : 'none',
+            padding                    : '20px'
 
```

Comment:
Very interesting! I see your point and I agree, I should refactor all modal windows to use this. The question is if I do that in another PR?